lib/reporter/magellan-reporter.js

```diff
@@ -142,19 +143,23 @@ Reporter.prototype.listenTo = function( testRun, test, source ) {
 							const slackUpload = new SlackUpload( config.get( 'slackTokenForScreenshots' ) );
 
 							try {
-								slackUpload.uploadFile( {
-									file: fs.createReadStream( `${screenshotDir}/${screenshotPath}` ),
-									title: `${screenshotPath} - # ${process.env.CIRCLE_BUILD_NUM}`,
-									channels: config.get( 'slackChannelForScreenshots' )
-								}, function( err ) {
-									if ( err ) {
-										slackClient.send( {
-											icon_emoji: ':a8c:',
-											text: `Build <https://circleci.com/gh/${process.env.CIRCLE_PROJECT_USERNAME}/${process.env.CIRCLE_PROJECT_REPONAME}/${process.env.CIRCLE_BUILD_NUM}|#${process.env.CIRCLE_BUILD_NUM}> Upload to slack failed: '${err}'`,
-											username: 'e2e Test Runner'
+								fs.createReadStream( `${screenshotDir}/${screenshotPath}` )
+									.pipe( pngitxt.get( 'url', function ( url ) {
```

Comment:
I wonder if this could be solved better but I'll need some input on this.

Reply:
I don't have any better suggestions, but perhaps @brbrr or @alisterscott does?

Reply:
I can't see any obvious ways to improve this either. 

It looks to be going against the '__spirit__' of the file's contents in general, but maybe the solution is just to abstract it a bit, and make it more readable (?)

```
function slackUploadFileCallback( err, data ) {
    if ( err ) {
        slackClient.send( {
            ...args
        } );
    }
}
function getStreamCallback() {
    return pngitxt.get( 'url', function ( url ) {
        slackUpload.uploadFile( { ...args }, slackUploadFileCallback );
    } );
}

fs.createReadStream( `${screenshotDir}/${screenshotPath}` )
    .pipe( getStreamCallback() );


```
None of the library methods returns a promise, I don't think that would make life easier in this case anyway.